### PR TITLE
claims: fix unicode handling

### LINKED ISSUE #44: Encoding issue
Apparently not showing correctly utf-8 chartset

selection_334

### PR DESCRIPTION
**Pull request by sameersbn**
closes #44

ref: https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding#The_Unicode_Problem

### CODE REVIEW

File: src/app/chart-details/chart-details-comments/chart-details-comments.component.ts [hunk 1]
```diff
@@ -39,7 +39,9 @@ export class ChartDetailsCommentsComponent implements OnInit {
     this.authService.loggedIn().subscribe(loggedIn => { this.loggedIn = loggedIn; });
     let userClaims = this.cookieService.get("ka_claims");
     if (userClaims) {
-      this.user = JSON.parse(atob(userClaims));
+      this.user = JSON.parse(decodeURIComponent(atob(userClaims).split('').map(function(c) {
```
Comment (prydonius) [inline src/app/chart-details/chart-details-comments/chart-details-comments.component.ts]: can you add a comment here and briefly explain with a link to https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding#The_Unicode_Problem? Or maybe separate this to a utility function that can be used in both places.

Comment (sameersbn) [reply src/app/chart-details/chart-details-comments/chart-details-comments.component.ts]: i've added `getUser` function under the authService.

File: src/app/shared/services/auth.service.ts [hunk 2]
```diff
@@ -24,9 +25,23 @@ export class AuthService {
...
 
+  /**
+   * Get the details of the logged in user
+   *
+   * @return {Observable} An observable object with the user info
+   */
+  getUser(): Observable<any> {
+    let userClaims = this.cookieService.get("ka_claims");
+    if (userClaims) {
+      return JSON.parse(decodeURIComponent(atob(userClaims).split('').map(function(c) {
```
Comment (prydonius) [inline src/app/shared/services/auth.service.ts]: I still think we should add a comment above this to explain why this is more complex than just `atob`, e.g.:

```
// parse as unicode string (see https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding#The_Unicode_Problem)
```

Comment (sameersbn) [reply src/app/shared/services/auth.service.ts]: done :+1:

**Pull request merged**